frontend/src/views/Market.vue

```diff
@@ -527,8 +586,28 @@ export default Vue.extend({
 
       this.selectedNftId = null;
       this.waitingMarketOutcome = false;
-      this.marketOutcome = 'Successfully changed price for '
-        +this.activeType+' '+results.nftID+' to '+this.convertWeiToSkill(results.newPrice)+' SKILL';
+      this.marketOutcome = `Successfully changed price for ${this.activeType} ${results.nftID} to ${this.convertWeiToSkill(results.newPrice)} SKILL`;
+    },
+
+    async updateNftListingTargetBuyer() {
+
+      this.marketOutcome = null;
+      if(this.selectedNftId === null) return;
+
+      let targetBuyer = await (this as any).$dialog.prompt({ title: `Sell ${this.activeType}`, text: 'Target Buyer Address (optional)' });
```

Comment:
Instead of doing `this as any`, the definition in `shims-vue.d.ts` should in theory make sure `$dialog` exists as a field already on `this` (the type of `this.$dialog` as per said file *is* just `any` anyway, but that's another issue). Would it be possible to remove the `as any` cast here?

Reply:
Tried it at first, it didn't let me. I just checked again why and it seems like 
`export default Vue.extend({` instead of `export default {` causes that issue. After trying to remove `Vue.extend(` part, `this` is broken in other scenarios, e.g. `this.contracts` - property contract does not exist on `this` component. Let me know if I should investigate that more and find a fix or leave as is.

Reply:
Going to try and get this merged today, so I'll take a look at it myself instead, thanks anyway.